src/tests/JIT/opt/And/IntAnd.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    class IntAnd
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void SideEffect()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
+
+            if ((byte)(x & y) == 0)
+            {
+                SideEffect();
+                return true;
+            }
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_CastByte_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
+
+            if ((byte)((byte)x & y) == 0)
+            {
+                SideEffect();
+                return true;
+            }
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_CastByte_CastByte_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
+
+            if ((byte)((byte)x & (byte)y) == 0)
+            {
+                SideEffect();
+                return true;
+            }
+            return false;
+        }
+
+        static int Main()
+        {
+            // No CastByte
+            if (!Test_UInt32_UInt32_And(0b10000000000000000000000000000000, 0b00000000000000000000000000000001))
```

Comment:
I suggest using the 0b1000_0000_0000 etc notation throughout these tests (especially if you're going to use binary rather than hex)

Reply:
That's fair, I can update that.